ecl/hql/hqldesc.cpp

```diff
@@ -113,15 +129,98 @@ static void expandRecordSymbolsMeta(IPropertyTree * metaTree, IHqlExpression * r
     }
 }
 
+void expandScopeMeta(IPropertyTree * meta, IHqlExpression * expr) {
+    if (expr->hasAttribute(virtualAtom))
+        meta->setPropBool("@virtual", true);
+
+    if (expr->hasAttribute(interfaceAtom)){
```

Comment:
That is true.  You will also need to be careful about not duplicating information when it is shared by multiple definitions - .g, only expanding the fields in a record once.  Other complications are the introduction of a distributed meta file generation - where a single file is generated for each source file.  I'm not sure what other problems that will create.